src/pymor/models/basic.py

```diff
@@ -29,6 +28,13 @@ def __init__(self, products=None, estimator=None, visualizer=None,
                 setattr(self, f'{k}_product', v)
                 setattr(self, f'{k}_norm', induced_norm(v))
 
+    def __repr__(self):
+        strings = [f'{self.__class__.__name__}(\n']
+        for parameter in self.__signature__.parameters:
+            strings.append(f'    {parameter}={getattr(self, parameter)!r},\n')
+        strings.append(')')
+        return ''.join(strings)
```

Comment:
If we want to have such a method, why only for `Models`? We might consider moving it to `BasicInterface`. Not sure what the consequences would be, however.